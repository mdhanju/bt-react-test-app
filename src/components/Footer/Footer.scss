footer {
    &#btrc-Footer {
        padding-bottom: 1.4rem;
        font-size: 1.5rem;
        color: $dark-indigo;
        @extend .hca-book;
    }
}
