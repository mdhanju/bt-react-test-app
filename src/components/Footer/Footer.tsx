import React from 'react'
import classnames from 'classnames'
import { FormattedMessage } from 'react-intl'

const Footer = ({ className, id, ...props }: Props): React.ReactElement => (
    <footer
        id={id}
        className={classnames(className, 'container-fluid')}
        {...props}
    >
        &copy; {new Date().getFullYear()}&nbsp;
        <FormattedMessage
            id="app.footername"
            description="Footer name showing Blue Triangle name"
        />
    </footer>
)

type Props = {
    id?: string;
    className?: string;
};

Footer.defaultProps = {
    id: 'btrc-Footer'
}

export default Footer
