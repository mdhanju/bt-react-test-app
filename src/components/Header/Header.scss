header {
    &#btrc-Header {
        background-color: $bt-blue;
        color: $white;
        height: 8.7rem;
        min-height: 8.7rem;
        display: flex;
        align-items: center;
        align-content: center;

        .container-fluid {
            display: flex;
            justify-content: space-between;
            align-items: flex-end;
            flex-wrap: nowrap;

            .logo {
                .mobile-logo {
                    width: 3.2rem;
                    height: 3.2rem;
                }
            }

            @include media-breakpoint-down(sm) {
                padding: 0;
            }
        }

        .logo {
            display: flex;
            align-items: baseline;

            @include media-breakpoint-down(sm) {
                align-items: center;
                margin-left: 1.6rem;
            }

            .app-name {
                margin-left: 2rem;
                font-size: 2rem;
                letter-spacing: -0.032rem;

                @include media-breakpoint-down(sm) {
                    font-size: 1.8rem;
                }
            }
        }
    }
}
