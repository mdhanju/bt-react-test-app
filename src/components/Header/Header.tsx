import React from 'react'
import classnames from 'classnames'
import desktopLogoSrc from '../../assets/images/logo-white.png'
import mobileLogoSrc from '../../assets/images/bt-icon.svg'
import { FormattedMessage } from 'react-intl'
import Image from '../Image'

const Header = ({ className, id, ...props }: Props): React.ReactElement => (
    <header id={id} className={classnames(className)} {...props}>
        <div className="container-fluid">
            <div className="logo">
                <Image
                    src={desktopLogoSrc}
                    alt="brand logo"
                    className="mobile-hide"
                />
                <Image
                    src={mobileLogoSrc}
                    alt="mobile brand logo"
                    className="mobile-logo desktop-hide"
                />
                <span className="app-name">
                    <FormattedMessage id="app.name" />
                </span>
            </div>
        </div>
    </header>
)

type Props = {
    id?: string;
    className?: string;
};

Header.defaultProps = {
    id: 'btrc-Header'
}

export default Header
