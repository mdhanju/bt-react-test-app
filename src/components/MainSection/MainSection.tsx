import React, { ReactNode } from 'react'
import classnames from 'classnames'

const MainSection = ({
    className,
    children,
    ...props
}: Props): React.ReactElement => (
    <section className={classnames(className)} {...props}>
        {children}
    </section>
)

type Props = {
    className?: string;
    children?: ReactNode;
};

export default MainSection
