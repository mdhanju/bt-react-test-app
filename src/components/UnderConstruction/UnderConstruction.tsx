import React from 'react'

import ContentHolder from '../ContentHolder'

const LoadingSpinner = (): React.ReactElement => {
    return <ContentHolder>Under Construction</ContentHolder>
}

export default LoadingSpinner
