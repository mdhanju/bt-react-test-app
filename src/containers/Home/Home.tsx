import React from 'react'

import ContentHolder from '../../components/ContentHolder'
import backgroundImg from '../../assets/images/home-bg.png'
import { connect } from 'react-redux'
import { State } from '../../store/reducers/rootReducer'
import { getFormData } from '../../store/actions/applicationInitActions'
import { ProfileAction } from '../../store/actions/applicationInitActions'

import { ThunkDispatch } from 'redux-thunk'

type Props = {
  profile?: any;
  isAppInitialized: string;
  getFormData: () => void;
};

class Home extends React.Component<Props> {
    componentDidMount(): void {
        (window as any).btPageName = "BT React Test Home Page"
        this.props.getFormData()
    }

    render(): React.ReactNode {
        if (!this.props.profile) {
            return <ContentHolder><h2>{'Loading...'}</h2></ContentHolder>
        }

        return <ContentHolder>
            <img src={backgroundImg} alt="background" />
        </ContentHolder>
    }
}


const mapStateToProps = (state: State): object => ({
    profile: state.appContext.profile,
    isAppInitialized: state.appContext.isAppInitialized
})

const mapDispatchToProps = (
    dispatch: ThunkDispatch<State, void, ProfileAction>
): object => ({
    getFormData: (): Promise<void> => dispatch(getFormData())
})

const HomeWithRedux: /* eslint-disable-line @typescript-eslint/no-explicit-any */ any = connect(
    mapStateToProps,
    mapDispatchToProps
)(Home)
export default HomeWithRedux
