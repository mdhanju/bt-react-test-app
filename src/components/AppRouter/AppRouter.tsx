import React, { Suspense } from 'react'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'

import Header from '../Header'
import Footer from '../Footer'
import MainSection from '../MainSection'
import ErrorPage from '../ErrorPage'
import LoadingSpinner from '../LoadingSpinner'

import Home from '../../containers/Home'
import ProfilePage from '../../containers/Profile'
import SampleForm from '../../containers/SampleForm'
import UnderConstruction from '../UnderConstruction'

type Props = {
    isAppInitialized: string;
};

const Routes = (): React.ReactElement => {
    return (
        <Suspense fallback={<LoadingSpinner />}>
            <Switch>
                <Route exact path="/" component={Home} />
                <Route exact path="/bt-test-home" component={Home} />
                <Route exact path="/addProfile" component={SampleForm} />
                <Route exact path="/profile" component={ProfilePage} />
                <Route exact path="/help" component={UnderConstruction} />
                <Route exact path="/error" component={ErrorPage} />
            </Switch>
        </Suspense>
    )
}

const AppRouter = ({ isAppInitialized }: Props): React.ReactElement => {
    return (
        <Router>
            <Header />
            <MainSection className="cp-main-section">
                {
                    ({
                        ERROR: <ErrorPage />,
                        INPROGESS: <LoadingSpinner />,
                        SUCCESS: <Routes />
                    } /* eslint-disable-line @typescript-eslint/no-explicit-any */ as any)[
                        isAppInitialized
                    ]
                }
            </MainSection>
            <Footer />
        </Router>
    )
}

export default AppRouter
