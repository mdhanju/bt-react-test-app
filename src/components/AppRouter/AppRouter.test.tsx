/* eslint-disable react/jsx-no-undef */

import React from 'react'
import { MemoryRouter } from 'react-router'
import Home from '../../containers/Home'
import UnderConstruction from '../UnderConstruction'
import Header from '../Header'
import Footer from '../Footer'
import MainSection from '../MainSection'

import AppRouter from '../AppRouter'

jest.mock('../../containers/Home', () => require('../../containers/Home/Home'))
jest.mock('../../containers/SampleForm', () =>
    require('../../containers/SampleForm/SampleForm')
)
jest.mock('../UnderConstruction', () =>
    require('../UnderConstruction/UnderConstruction')
)

describe('<AppRouter />', () => {
    it('should render initial layout - INPROGRESS', () => {
        const wrapper = shallow(
            <Intl locale="en">
                <AppRouter isAppInitialized="INPROGRESS" />
            </Intl>
        )
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
    })

    it('should render initial layout - ERROR', () => {
        const wrapper = shallow(
            <Intl locale="en">
                <AppRouter isAppInitialized="ERROR" />
            </Intl>
        )
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
    })

    it('should render initial layout - SUCCESS', () => {
        const wrapper = shallow(
            <Intl locale="en">
                <AppRouter isAppInitialized="SUCCESS" />
            </Intl>
        )
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
    })

    it('should render When - INPROGRESS', () => {
        const wrapper = render(
            <Intl locale="en">
                <MemoryRouter initialEntries={['/random']}>
                    <AppRouter isAppInitialized="INPROGRESS" />
                </MemoryRouter>
            </Intl>
        )
        expect(wrapper.text()).toBe(
            `Sample App© ${new Date().getFullYear()}\u00a0Blue Triangle, Inc.`
        )
        expect(wrapper.html()).toContain(
            '<img src="logo-white.png" class="mobile-hide" alt="brand logo">'
        )
        expect(wrapper.html()).toContain(
            '<img src="bt-icon.svg" class="mobile-logo desktop-hide" alt="mobile brand logo">'
        )
        expect(wrapper.html()).toContain(
            '<span class="app-name">Sample App</span>'
        )
    })

    it('should render When - ERROR', () => {
        const wrapper = render(
            <Intl locale="en">
                <MemoryRouter initialEntries={['/random']}>
                    <AppRouter isAppInitialized="ERROR" />
                </MemoryRouter>
            </Intl>
        )
        expect(wrapper.text()).toBe(
            'Sample AppHome - Sample FormAn error occured while loading configuration.' +
                '© 2020 Blue Triangle, Inc.'
        )
        expect(wrapper.html()).toContain(
            '<img src="logo-white.png" class="mobile-hide" alt="brand logo">'
        )
        expect(wrapper.html()).toContain(
            '<img src="bt-icon.svg" class="mobile-logo desktop-hide" alt="mobile brand logo">'
        )
        expect(wrapper.html()).toContain(
            '<span class="app-name">Sample App</span>'
        )
    })

    it('should render When - SUCCESS', () => {
        const wrapper = mount(
            <Intl locale="en">
                <MemoryRouter initialEntries={['/random']}>
                    <AppRouter isAppInitialized="SUCCESS" />
                </MemoryRouter>
            </Intl>
        )
        expect(wrapper.text()).toBe(
            `Sample App© ${new Date().getFullYear()}\u00a0Blue Triangle, Inc.`
        )
        expect(wrapper.html()).toContain(
            '<img src="logo-white.png" class="mobile-hide" alt="brand logo">'
        )
        expect(wrapper.html()).toContain(
            '<img src="bt-icon.svg" class="mobile-logo desktop-hide" alt="mobile brand logo">'
        )
        expect(wrapper.html()).toContain(
            '<span class="app-name">Sample App</span>'
        )
    })

    it('should mount Home on - SUCCESS', () => {
        const wrapper = mount(
            <Intl locale="en">
                <MemoryRouter initialEntries={['/']}>
                    <AppRouter isAppInitialized="SUCCESS" />
                </MemoryRouter>
            </Intl>
        )
        expect(wrapper.find(Header)).toHaveLength(1)
        expect(wrapper.find(Footer)).toHaveLength(1)
        expect(wrapper.find(MainSection)).toHaveLength(1)
        expect(wrapper.find(MainSection).find(Home)).toHaveLength(1)
        expect(wrapper.find(MainSection).find(UnderConstruction)).toHaveLength(
            0
        )
    })
    it('should mount Help on - SUCCESS', () => {
        const wrapper = mount(
            <Intl locale="en">
                <MemoryRouter initialEntries={['/help']}>
                    <AppRouter isAppInitialized="SUCCESS" />
                </MemoryRouter>
            </Intl>
        )
        expect(wrapper.find(Header)).toHaveLength(1)
        expect(wrapper.find(Footer)).toHaveLength(1)
        expect(wrapper.find(MainSection)).toHaveLength(1)
        expect(wrapper.find(MainSection).find(Home)).toHaveLength(0)
        expect(wrapper.find(MainSection).find(UnderConstruction)).toHaveLength(
            1
        )
    })
})
