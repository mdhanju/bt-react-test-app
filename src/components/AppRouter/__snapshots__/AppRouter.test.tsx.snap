// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> should render initial layout - ERROR 1`] = `
Array [
  <IntlProvider
    defaultFormats={Object {}}
    defaultLocale="en"
    formats={Object {}}
    locale="en"
    messages={
      Object {
        "app.footername": "Blue Triangle, Inc.",
        "app.name": "Sample App",
        "err.page.system.error": "An error occured while loading configuration.",
        "errorboundry.error": "System error encountered. Try again...",
        "nav.help": "Help",
        "nav.home": "Home",
      }
    }
    onError={[Function]}
    textComponent={Symbol(react.fragment)}
  >
    <AppRouter
      isAppInitialized="ERROR"
    />
  </IntlProvider>,
]
`;

exports[`<AppRouter /> should render initial layout - INPROGRESS 1`] = `
Array [
  <IntlProvider
    defaultFormats={Object {}}
    defaultLocale="en"
    formats={Object {}}
    locale="en"
    messages={
      Object {
        "app.footername": "Blue Triangle, Inc.",
        "app.name": "Sample App",
        "err.page.system.error": "An error occured while loading configuration.",
        "errorboundry.error": "System error encountered. Try again...",
        "nav.help": "Help",
        "nav.home": "Home",
      }
    }
    onError={[Function]}
    textComponent={Symbol(react.fragment)}
  >
    <AppRouter
      isAppInitialized="INPROGRESS"
    />
  </IntlProvider>,
]
`;

exports[`<AppRouter /> should render initial layout - SUCCESS 1`] = `
Array [
  <IntlProvider
    defaultFormats={Object {}}
    defaultLocale="en"
    formats={Object {}}
    locale="en"
    messages={
      Object {
        "app.footername": "Blue Triangle, Inc.",
        "app.name": "Sample App",
        "err.page.system.error": "An error occured while loading configuration.",
        "errorboundry.error": "System error encountered. Try again...",
        "nav.help": "Help",
        "nav.home": "Home",
      }
    }
    onError={[Function]}
    textComponent={Symbol(react.fragment)}
  >
    <AppRouter
      isAppInitialized="SUCCESS"
    />
  </IntlProvider>,
]
`;
